workbench-core/infrastructure/src/workbenchAppRegistry.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as appreg from '@aws-cdk/aws-servicecatalogappregistry-alpha';
+import { SharePermission } from '@aws-cdk/aws-servicecatalogappregistry-alpha';
+import { Aws, CfnMapping, Fn, Stack } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { applyTag } from './workbenchApplyTags';
+
+export interface WorkbenchAppRegistryProps {
+  solutionId: string;
+  solutionName: string;
+  solutionVersion: string;
+  appRegistryApplicationName: string;
+  applicationType: string;
+  attributeGroupName: string;
+  accountIds?: string[];
+}
+
+export class WorkbenchAppRegistry extends Construct {
+  private _solutionId: string;
+  private _solutionName: string;
+  private _solutionVersion: string;
+  private _appRegistryApplicationName: string;
+  private _applicationType: string;
+  private _attributeGroupName: string;
+  private _accountIds: string[];
+  private readonly _registryApplication: appreg.Application;
+  private readonly _appRegMap: CfnMapping;
+
+  public constructor(scope: Construct, id: string, props: WorkbenchAppRegistryProps) {
+    super(scope, id);
```

Comment:
do we have cases where the scope will be different from the stack ?

Reply:
if not we can get the id from the scope

Reply:
@FernandoAranda For our use case, when we integrate App-Registry it will be associated with atleast one stack so why not pass it as a scope. But we can revisit this if this becomes a problem.